sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/PlanController.java

```diff
@@ -29,55 +38,69 @@
  */
 @RestController
 @RequiredArgsConstructor
-@RequestMapping(value = "${api.version}/sponsorship/plans",
+@RequestMapping(value = "${api.version.path}/plans",
     consumes = { MediaType.APPLICATION_JSON_UTF8_VALUE }, produces = { MediaType.APPLICATION_JSON_UTF8_VALUE })
 @Api(value = "Register a new sponsorship plan")
 public class PlanController
 {
   private final PlanService planService;
 
 
-  @SuppressWarnings("unused")
   @ApiOperation(value = "Register the plan")
-  @PostMapping
   @ApiResponses(value = {
-      @ApiResponse(code = 200, message = "OK", response = PlanRegistered.class),
-      @ApiResponse(code = 401, message = "Failed to register the plan", response = PlanRegistered.class)
+      @ApiResponse(code = 200, message = "OK"),
+      @ApiResponse(code = 401, message = "Failed to register the plan")
   })
-  PlanRegistered register(@Valid @RequestBody PlanRegistrationRequest request)
+  @PostMapping
+  PlanRegistrationResponse register(@Valid @RequestBody PlanRegistrationRequest request)
   {
-    final String randomId = "234";
-    final Plan plan = Plan.builder()
-        .id(randomId)
-        .category(request.getCategory())
-        .sponsorId(request.getSponsorEmail())
-        .build();
-    return new PlanRegistered(plan.getId());
+    return new PlanRegistrationResponse(planService.createPlan(request.getCategory(), request.getSponsorEmail()));
   }
 
 
-  @SuppressWarnings("unused")
   @ApiOperation(value = "Cancel the plan")
-  @PutMapping("/{id}/cancellation")
   @ApiResponses(value = {
-      @ApiResponse(code = 200, message = "OK", response = PlanCancellationResponse.class),
-      @ApiResponse(code = 401, message = "Failed to cancel the plan", response = PlanCancellationResponse.class)
+      @ApiResponse(code = 200, message = "OK"),
+      @ApiResponse(code = 401, message = "Failed to cancel the plan")
   })
-  PlanCancellationResponse cancel(@ApiParam(value = "Id of the plan to cancel", required = true) @PathVariable String id)
+  @DeleteMapping("/{id}")
+  void cancel(@ApiParam(value = "Id of the plan to cancel", required = true) @PathVariable Long id)

```

Comment:
In my point of view 'cancel' means that you change a status of plan -> it remains in DB but cannot be applied/active. Here you just removes it.

Reply:
If plan is removed from DB then you have to decide what to do with related entries, for example sponsors who participated with this plan and have references on it.

Reply:
It is designed in a different way. Sponsor doesn't have reference to plan,  but each plan associated with a sponsor has a reference to its sponsor